plugins/storage/volume/storpool/src/main/java/org/apache/cloudstack/storage/datastore/driver/StorPoolPrimaryDataStoreDriver.java

```diff
@@ -431,6 +439,54 @@ public void deleteAsync(DataStore dataStore, DataObject data, AsyncCompletionCal
         callback.complete(res);
     }
 
+    private void tryToSnapshotVolumeBeforeDelete(VolumeInfo vinfo, String name, SpConnectionDesc conn) {
+        Integer deleteAfter = StorPoolConfigurationManager.DeleteAfterInterval.value();
+        if (deleteAfter != null && deleteAfter > 0 && vinfo.getPassphraseId() == null) {
+            createTemporarySnapshot(vinfo, name, deleteAfter, conn);
+        } else {
+            StorPoolUtil.spLog("The volume [%s] is not marked to be snapshoted. Check the global setting `storpool.delete.after.interval` or the volume is encrypted [%s]", name, deleteAfter, vinfo.getPassphraseId() != null);
```

Comment:
```suggestion
            StorPoolUtil.spLog("The volume [%s] is not marked to be snapshotted. Check the global setting `storpool.delete.after.interval` or the volume is encrypted [%s]", name, deleteAfter, vinfo.getPassphraseId() != null);
```

Reply:
Thanks, @sureshanaparti for the review! Fixes are provided in the following commit